packages/calcite-components/src/components/modal/modal.scss

```diff
@@ -18,7 +18,7 @@
 }
 
 :host {
-  @apply absolute flex inset-0 opacity-0 z-overlay;
+  @apply fixed flex inset-0 opacity-0 z-overlay;
```

Comment:
Would we need to do this for the dialog component as well?

Reply:
We're going to test drive this with `modal` first. If everything works out smoothly, we could apply to other components before 4.0 since it wouldn't be breaking (see https://github.com/Esri/calcite-design-system/pull/9545#discussion_r1635722740).